src/twisted/web/_flatten.py

```diff
@@ -167,6 +166,39 @@ def escapedCDATA(data: Union[bytes, str]) -> bytes:
     return data.replace(b"]]>", b"]]]]><![CDATA[>")
 
 
+def _hasMSOComments(html_string: Union[str, bytes]) -> bool:
+    if isinstance(html_string, bytes):
```

Comment:
this branch is not covered... it looks like html_string is always bytes in the tests.

Reply:
good point, if `data` is a string then it will be encoded to bytes using the utf-8 encoding, that means that the data argument passed to `_hasMSOComments` is always of type bytes. I will fix it.